utils/compression/zstd_compressor.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package compression
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/DataDog/zstd"
+)
+
+var _ Compressor = (*zstdCompressor)(nil)
+
+func NewZstdCompressor(maxSize int64) Compressor {
+	return &zstdCompressor{
+		maxSize: maxSize,
```

Comment:
Mega nit, but now that we do `maxSize + 1` can we handle that overflow like we do with `gzip`? I probably wouldn't care about it... but since we handle the error in gzip I feel like we should be consistent.